hack/client/generate-client.sh

```
@@ -0,0 +1,40 @@
+  --output-dir $(pwd)/${OUTPUT_DIR} \
+  --clientset-name clientset \
+  --versioned-name versioned \
+  --listers-name listers \
+  --applyconfig-name applyconfiguration \
+  --boilerplate ${BOILERPLATE_FILE} \
+  --with-watch
+
+# Recover the original client directory if the generation failed.
+if [ $? -ne 0 ]; then
+  echo "Client generation failed, recovering the original client directory."
+  mv ${BACKUP_DIR}/client $(pwd)/${OUTPUT_DIR}
```

Shellcheck (0.10.0)

[warning] Quote this to prevent word splitting.

(SC2046)